packages/scandipwa/src/route/ProductPage/ProductPage.component.js

```diff
@@ -104,31 +104,118 @@ export class ProductPage extends PureComponent {
         );
     }
 
-    renderAdditionalSections() {
+    isProductInformationTabEmpty() {
+        const { dataSource } = this.props;
+
+        return !(dataSource
+            && dataSource.description
+            && dataSource.description.html
+            && dataSource.description.html.length);
+    }
+
+    isProductAttributesTabEmpty() {
+        const { dataSource } = this.props;
+
+        return !(dataSource
+                && dataSource.attributes
+                && !(Object.keys(dataSource.attributes).length === 0));
+    }
+
+    getTabNames() {
```

Comment:
This is some primitive code, can you instead change it to be a map of validation / title / render? Will look much better